core/src/main/java/org/apache/iceberg/util/ArrayUtil.java

```diff
@@ -59,6 +61,14 @@ public static List<Long> toLongList(long[] longs) {
     }
   }
 
+  public static List<Long> toUnmodifiableLongList(long[] longs) {
+    if (longs != null) {
+      return Collections.unmodifiableList(Longs.asList(longs));
```

Comment:
I am using `Longs.asList()` from Guava instead of Stream + collect as it wraps the array directly into a special collection type called `LongArrayAsList`, which should generate less garbage and be faster. I noticed these calls in flamegraphs.

<img width="2551" alt="image" src="https://github.com/apache/iceberg/assets/6235869/8cc8e93e-7c12-4519-8ff4-3f3e42af924a">

Reply:
I believe `Arrays.asList()` works well for object arrays, we are dealing with `long[]` here.